moto/iam/models.py

```diff
@@ -569,6 +569,16 @@ def get_role(self, role_name):
                 return role
         raise IAMNotFoundException("Role {0} not found".format(role_name))
 
+    def delete_role(self, role_name):
```

Comment:
@xoen Good idea! I had to keep it a little bit complex because the roles are indexed by id rather than name but the method is much shorter now. Thanks for your help